lib/_tls_wrap.js

```diff
@@ -1084,6 +1088,8 @@ function tlsConnectionListener(rawSocket) {
 
   socket[kErrorEmitted] = false;
   socket.on('close', onSocketClose);
+  // Handle ECONNRESET
+  socket.on('error', onSocketError);
```

Comment:
Isn't this `'error'` event already handled by https://github.com/nodejs/node/blob/0ca861745ad47f1b02dbb4ea30b0d7d4328f0719/lib/_tls_wrap.js#L510?

Reply:
Yes, you are right.
Yet, when the ECONNRESET error is triggered, there is no error handler.
I see there are a couple of cases where the handler is removed just before closing, so I guess it must have been one of those cases. I will dig further.

Reply:
That listener is removed when the `'secureConnection'` event is emitted but at that point it is user responsibility to handle the error.

Reply:
The user has no access to the underlying socket?

Reply:
Yes they do, it's the "socket" that is passed to the handlers of the [`'secureConnection'`](https://nodejs.org/api/tls.html#tls_event_secureconnection) event.